scripts/diagnostics.sh

```diff
@@ -8,138 +8,187 @@
-        write_to_stdout "\n\$CC not set"
-    fi
+function print_failure_and_exit()
+{
+    print_environment
+    printf "\nThere was an error while $(print_red "$@")\033[0m.
 
-    # Static
-    which mpicc &> /dev/null
-    if [ "$?" -eq 0 ]; then
-        write_to_stdout "\nMPICC:\nwhich mpicc:\n\t`which mpicc`\nmpicc -show:\n\t`mpicc -show`\n\nCOMPILER `echo $(mpicc -show | cut -d\  -f1)`:\n`$(mpicc -show | cut -d\  -f1) --version`"
-    else
-        write_to_stdout "\nMPICC WRAPPER NOT FOUND"
-    fi
+Please report the entirety of the output of diagnostics on
+this terminal to either YOUR existing post OR a new post on
+https://github.com/idaholab/moose/discussions\n"
+    exit 1
 }
 
-diagnose_modules() {
-    type module &> /dev/null
-    if [ "$?" -eq 0 ]; then
-        write_to_stdout "\nMODULES:\n`module list 2>&1`"
-    else
-        write_to_stdout "\nMODULES NOT AVAILABLE"
-    fi
+function print_sep()
+{
+    printf '\n'
+    printf '#%.0s' {1..75}
+    printf '\n'
 }
 
-diagnose_root() {
-    if [ $(id -u) = 0 ]; then write_to_stdout "RUNNING AS ROOT"; fi
+function print_red()
+{
+    printf "\033[1;31m$@\033[0m"
 }
 
-diagnose_environment() {
-    write_to_stdout "\nENVIRONMENT:\n`env | sort`"
+function print_orange()
+{
+    printf "\033[38;5;202m$@\033[0m"
 }
 
-diagnose_cpus() {
-    if [ `uname` == "Darwin" ]; then
-        CPUS=`/usr/sbin/sysctl -n hw.ncpu`
-    else
-        CPUS=`cat /proc/cpuinfo | grep processor | wc -l`
-    fi
-    write_to_stdout "\nCPU Count: `echo $CPUS`"
+function print_green()
+{
+    printf "\033[1;32m$@\033[0m"
 }
 
-diagnose_memory() {
+function create_tmp()
+{
+    if [ -n "$CTMP_DIR" ]; then echo $CTMP_DIR; return; fi
     if [ `uname` == "Darwin" ]; then
-        PAGE_SIZE=`vm_stat | grep "page size of" | sed 's/[^0-9]*//g'`
-        MEMORY_FREE=`vm_stat | grep free | awk -v page_size=$PAGE_SIZE '{ print (page_size * $3) / 1048576}'`
+        local TMP_DIR=${TMP_DIR:-`mktemp -d -t MOOSE`}
     else
-        MEMORY_FREE=`awk '/MemFree/ { printf "%.3f \n", $2/1024 }' /proc/meminfo`
+        local TMP_DIR=${TMP_DIR:-`mktemp -d -t MOOSE.XXXXXXXX`}
     fi
-    write_to_stdout "\nMemory Free: `echo $MEMORY_FREE` MB"
+    echo $TMP_DIR
 }
 
-diagnose_arch() {
-    if [ `uname` == "Darwin" ]; then
-        SYSTEM=`uname -v`
-    else
-        which lsb_release &> /dev/null
-        if [ "$?" -eq 0 ]; then
-            SYSTEM="`lsb_release -a`"
-        else
-            if [ -f /etc/system-release ]; then
-                SYSTEM="`cat /etc/system-release`"
-            else
-                SYSTEM="Not Available"
-            fi
-        fi
-    fi
-    write_to_stdout "\nSystem Arch: `echo $SYSTEM`"
+function print_help()
+{
+    args=("-h|--help"\
+          "-f|--full-build"\
+          "-r|--run-checks"\
+          "-a|--active-environment"\
+          "-c|--conda-channel")
+
+    args_about=("Print this message and exit."\
+                "Build everything: PETSc, libMesh, WASP, MOOSE."\
```

Comment:
```suggestion
                "Test build everything using conda: PETSc, libMesh, WASP, MOOSE."\
```

Reply:
well actually, not using conda.
why ? should there be two options?
--full-build-conda
--full-build-source ?